### PR TITLE
FeetInches to architectural string

### LINKED ISSUE #1075: FeetInches.ToString as architectural format
**Is your feature request related to a problem? Please describe.**
Not really

**Describe the solution you'd like**
It would be nice if FeetInches had a method to output a string in architectural format (including fractional inches). Using the example from below, it would output a string like `5' - 4 1/2"`.  

https://github.com/angularsen/UnitsNet/blob/ebd3275b84991f5d90113a03fb569cddd655c86e/UnitsNet/CustomCode/Quantities/Length.extra.cs#L235

**Describe alternatives you've considered**
An extension method works fine, and I'm happy to do this if it's not desired in this codebase but figured there may be others wishing for similar functionality.

**Additional context**
This is a pretty common notation in the US in certain contexts. I'm not sure if (the few) other cultures that use feet and inches have a different architectural format, so not sure how CultureInfo might play a role if any.

Related issue (never implemented AFIAK) #676. This would have to be implemented at the same time, so another thing to consider is how to deal with the fractional precision for rounding. Maybe via an argument, like an integer for the denominator?

I can contribute if this is wanted. I'm assuming we wouldn't want to override `ToString()` or any of it's overloads since that's already established. Maybe it'd be like `ToArchitecturalString(int fractionDenom)` ?

### PR DESCRIPTION
**Pull request by lwestfall**
Fixes #1075. 

Also fixes broken image URL in readme from a long time ago :)

My coverage gutters weren't working locally so not 100% sure the new method has full coverage. Will wait for codecov to do its thing and will add test cases as needed.

### CODE REVIEW

File: UnitsNet/CustomCode/Quantities/Length.extra.cs [hunk 1]
```diff
@@ -236,5 +236,70 @@ public string ToString(IFormatProvider? cultureInfo)
             // So inches are rounded when converting from base units to feet/inches.
             return string.Format(cultureInfo, "{0:n0} {1} {2:n0} {3}", Feet, footUnit, Math.Round(Inches), inchUnit);
         }
+
+        /// <summary>
+        ///     Outputs feet and inches on the format: {feetValue}' - {inchesValueIntegral}[ {inchesValueFractional}]"
+        ///     The inches are rounded to the nearest fraction of the fractionDenominator argument and reduced over the greatest common divisor.
+        /// </summary>
+        /// <param name="fractionDenominator">The maximum precision to express the rounded inch fraction part</param>
+        /// <example>Length.FromFeetInches(3,2).ToArchitecturalString(32).ToString() outputs: "3 ft 2 in"</example>
```
Comment (angularsen) [inline UnitsNet/CustomCode/Quantities/Length.extra.cs:245]: Shouldn't the example output string be `3' - 2"` ?

File: UnitsNet/CustomCode/Quantities/Length.extra.cs [hunk 2]
```diff
@@ -236,5 +236,70 @@ public string ToString(IFormatProvider? cultureInfo)
             // So inches are rounded when converting from base units to feet/inches.
             return string.Format(cultureInfo, "{0:n0} {1} {2:n0} {3}", Feet, footUnit, Math.Round(Inches), inchUnit);
         }
+
+        /// <summary>
+        ///     Outputs feet and inches on the format: {feetValue}' - {inchesValueIntegral}[ {inchesValueFractional}]"
+        ///     The inches are rounded to the nearest fraction of the fractionDenominator argument and reduced over the greatest common divisor.
+        /// </summary>
```
Comment (angularsen) [inline UnitsNet/CustomCode/Quantities/Length.extra.cs:243]: Missing `/` before `{inchesValueIntegral}`.
Should probably also give a short description that the fractional will be omitted if it is `1`.

File: UnitsNet/CustomCode/Quantities/Length.extra.cs [hunk 3]
```diff
@@ -236,5 +236,70 @@ public string ToString(IFormatProvider? cultureInfo)
...
+        /// <param name="fractionDenominator">The maximum precision to express the rounded inch fraction part</param>
+        /// <example>Length.FromFeetInches(3,2).ToArchitecturalString(32).ToString() outputs: "3 ft 2 in"</example>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when fractionDenominator is &lt; 1 (use 1 to round to nearest integer, with no fraction)</exception>
```
Comment (angularsen) [inline UnitsNet/CustomCode/Quantities/Length.extra.cs:246]: I would move the `use 1 to round to nearest integer, with no fraction` description to `fractionDenominator` parameter xmldoc, so it helps explain what values you can give.

File: UnitsNet/CustomCode/Quantities/Length.extra.cs [hunk 4]
```diff
@@ -236,5 +236,70 @@ public string ToString(IFormatProvider? cultureInfo)
...
+        public string ToArchitecturalString(int fractionDenominator)
+        {
+            if (fractionDenominator < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(fractionDenominator), "Denominator for fractional inch cannot be zero");
```
Comment (angularsen) [inline UnitsNet/CustomCode/Quantities/Length.extra.cs:251]: To be more precise, maybe change it to `"Denominator for fractional inch must be greater than zero."`

Comment (angularsen) [reply UnitsNet/CustomCode/Quantities/Length.extra.cs:251]: You could also put this text in the `ArgumentOutOfRangeException` xmldoc.

File: UnitsNet/CustomCode/Quantities/Length.extra.cs [hunk 5]
```diff
@@ -236,5 +236,70 @@ public string ToString(IFormatProvider? cultureInfo)
...
+                        if (a > b)
+                            a %= b;
+                        else
+                            b %= a;
+                    }
+
+                    return a | b;
+                }
+
+                var gcd = greatestCommonDivisor((int)Math.Abs(numerator), fractionDenominator);
```
Comment (angularsen) [inline UnitsNet/CustomCode/Quantities/Length.extra.cs:285]: Nitpick, but `int gcd = ` would be slightly easier to read here.

File: UnitsNet/CustomCode/Quantities/Length.extra.cs (hunk 1 above)
Comment (angularsen) [reply UnitsNet/CustomCode/Quantities/Length.extra.cs:245]: I would love a few more examples to help illustrate how different `fractionDenominator` values determines the output for the same input.

For example:
```xmldoc
<example>
<code>
var length = Length.FromFeetInches(3, 2.6);
length.ToArchitecturalString(1)   => 3' - 3"
length.ToArchitecturalString(2)   => 3' - 2 1/2"
length.ToArchitecturalString(4)   => 3' - 2 1/2"
length.ToArchitecturalString(8)   => 3' - 2 5/8"
length.ToArchitecturalString(16)  => 3' - 2 5/8"
length.ToArchitecturalString(32)  => 3' - 2 19/32"
length.ToArchitecturalString(128) => 3' - 2 77/128"
</code>
</example>
```

File: UnitsNet.Tests/CustomCode/LengthTests.cs [hunk 6]
```diff
@@ -233,5 +233,24 @@ public static void InverseReturnsReciprocalLength(double value, double expected)
 
             Assert.Equal(expected, inverseLength.InverseMeters);
         }
+
+        [Theory]
+        [InlineData(3, 2.563, 16, "3' - 2 9/16\"")]
+        [InlineData(3, 2.563, 32, "3' - 2 9/16\"")]
+        [InlineData(3, 2, 16, "3' - 2\"")]
+        [InlineData(3, 2.5, 1, "3' - 2\"")]
+        [InlineData(0, 2, 32, "2\"")]
```
Comment (angularsen) [inline UnitsNet.Tests/CustomCode/LengthTests.cs:242]: Nice tests.

Can we also add these test values from my example in another PR comment? They helped me understand how it works better. 

Adapt the example to [InlineData].

```
var length = Length.FromFeetInches(3, 2.6);
length.ToArchitecturalString(1)   => 3' - 3"
length.ToArchitecturalString(2)   => 3' - 2 1/2"
length.ToArchitecturalString(4)   => 3' - 2 1/2"
length.ToArchitecturalString(8)   => 3' - 2 5/8"
length.ToArchitecturalString(16)  => 3' - 2 5/8"
length.ToArchitecturalString(32)  => 3' - 2 19/32"
length.ToArchitecturalString(128) => 3' - 2 77/128"
```

**Review by angularsen — CHANGES_REQUESTED** [review]
Thanks! This looks great, just a few minor suggestions.

**Comment by lwestfall** [comment]
Thanks for the feedback! I'll address these and try to push by the end of the week, work is a little busy for me the next couple of days.

**Comment by lwestfall** [comment]
Fixes pushed! I think I addressed all comments, and I left them open so you can re-review. Let me know if you find anything else

**Comment by angularsen** [comment]
Awesome, great work! The nuget should be out in a few minutes.

[Release UnitsNet/4.138.0 · angularsen/UnitsNet](https://github.com/angularsen/UnitsNet/releases/tag/UnitsNet%2F4.138.0)

**Comment by angularsen** [comment]
Next hour, build times are very slow lately.

**Pull request merged**